### PR TITLE
Updated call to civicrm_views_custom_data_cache as per CRM-18776

### PR DESCRIPTION
**Pull request by adixon**
- [CRM-18776: Views and non-Contact entities with custom field groups](https://issues.civicrm.org/jira/browse/CRM-18776)

### CODE REVIEW

File: modules/views/components/civicrm.core.inc [hunk 1]
```diff
@@ -2711,9 +2711,10 @@ function _civicrm_core_data(&$data, $enabled) {
 
 
   while ($dao->fetch()) {
-    // call getTree using $dao->id as groupID and $dao->extends as entityType
-
-    $data = civicrm_views_custom_data_cache($data, $dao->extends, $dao->id, $dao->extends_entity_column_value);
+    // call getTree using $dao->id as groupID, $dao->extends as entityType, with possible subtypes in $dao->extends_entity_column_value
+    $contact_subtypes = (('Contact' != $dao->extends) || empty($dao->extends_entity_column_value)) ? NULL : array_filter(explode(CRM_Core_DAO::VALUE_SEPARATOR, $dao->extends_entity_column_value));
```
Comment (universalhandle) [inline modules/views/components/civicrm.core.inc]: There's a lot happening on this line! This is a bit easier to read/maintain:

``` php
    $extendsContactSubtype = (('Contact' === $dao->extends) && !empty($dao->extends_entity_column_value));
    $contact_subtypes = !$extendsContactSubtype ? NULL : array_filter(explode(CRM_Core_DAO::VALUE_SEPARATOR, $dao->extends_entity_column_value));
```

Also, I think civilint will reject `$contact_subtypes`. `$contactSubtypes` should work, though.

**Comment by JoeMurray** [comment]
Alan, as Release Manager this month, I'm trying to recruit people to help pare down the backlog of 91 PRs, some going back to last summer. I'm wondering if you would be able to help QA another PR if I got someone to QA this PR?

**Comment by adixon** [comment]
You can always try me for QAing a PR if you think I have any useful
expertise.

I think we've got enough people looking at this one already.
- Alan

On Mon, Jun 13, 2016 at 12:02 PM, Joe Murray notifications@github.com
wrote:

> Alan, as Release Manager this month, I'm trying to recruit people to help
> pare down the backlog of 91 PRs, some going back to last summer. I'm
> wondering if you would be able to help QA another PR if I got someone to QA
> this PR?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> https://github.com/civicrm/civicrm-drupal/pull/386#issuecomment-225626741,
> or mute the thread
> https://github.com/notifications/unsubscribe/AAKUGLvz5GrCWzyn7O_6BL542oZuTs_Eks5qLX8AgaJpZM4IyUQK
> .

## 

Alan Dixon, Web Developer

Drupal and CiviCRM for Canadian nonprofits.
http://blackflysolutions.ca/

**Comment by colemanw** [comment]
@adixon I think Frank's suggestion is a good one. Could you make that change please and then we can get this merged in.

**Comment by adixon** [comment]
Done.

**Comment by JKingsnorth** [comment]
Please also see CRM-19545

**Pull request merged**